File: 2/1_Cuatrimestre/Programacion_y_Diseño_Orientado_a_Objetos/JavaFinal/NapakalakiJavaSinInterfaz/src/NapakalakiGame/BadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author javigomez
 */
public abstract class BadConsequence {
    static int MAXTREASURES=10;
    private String text;
    private int levels;
    
    public abstract boolean isEmpty();
    public int getLevels(){
        return levels;
    }
    
    public abstract void substractVisibleTreasure(Treasure t);
    public abstract void substratHiddenTreasure(Treasure t);
    
    public abstract BadConsequence adjustToFitTreasureLists(ArrayList <Treasure> v, ArrayList <Treasure> h);
    public String toString(){
        return text+"   niveles->"+levels;
    }
    
    ////////////////////////////////////////////////////////////////////////////////////////////////////
    public String getText(){
            return text;
    }
    public BadConsequence(String texto, int levels){
        this.text=texto;
        this.levels=levels;
    }
}

File: 2/1_Cuatrimestre/Programacion_y_Diseño_Orientado_a_Objetos/RubyFinal/NapakalakiRuby/lib/BadConsequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'Monster.rb'
require_relative 'TreasureKind.rb'
module NapakalakiGame
  

class BadConsequence
  @@MAXTREASURES=10
  def BadConsequence.MAXTREASURES
   return @@MAXTREASURES
  end
  attr_accessor :text, :levels, :nVisibleTreasures, :nHiddenTreasures, :specificVisibleTreasures, :specificHiddenTreasures, :death
  def  initialize(aText, someLevels) 
    @text = aText;
    @levels = someLevels
  end
 
  
  
  def to_s()
    "#{@text}, niveles->#{@levels}"
  end

end

end
